src/components/shared/TokenAmountDisplay.tsx

```diff
@@ -1,16 +1,18 @@
 import { formatNumber } from "@/lib/utils";
 import { BigNumberish } from "ethers";
 import React, { useMemo } from "react";
+import Tenant from "@/lib/tenant/tenant";
+const { token } = Tenant.current();
```

Comment:
This is interesting. You made the token a class-level variable instead of the instance-level. I have never done things this way but I guess it works 🤷‍♂️